core/time/shared/tz.rb

```diff
@@ -0,0 +1,45 @@
+describe :time_tz, shared: true do
+  describe "TZ env variable" do
+    # We don't check #zone because it returns planform specific abbreviations
+    # e.g. "a/H" or "Asi" for "Asia/Hong_Kong" or "Coordinated Universal Time" for "UTC"
+    # on Windows
+    #
+    # see https://bugs.ruby-lang.org/issues/13591#note-11
+
+    before do
+      @tz = ENV["TZ"]
+
+      @get_time = @object
+    end
+
+    after do
+      ENV["TZ"] = @tz
+    end
+
+    it "returns time in timezone specified by TZ env variable" do
+      ENV["TZ"] = "Asia/Hong_Kong"
+      @get_time.call.utc_offset.should == 8*60*60 # HKT, +08:00
+
+      ENV["TZ"] = "Africa/Dakar"
+      @get_time.call.utc_offset.should == 0 # "GMT"
```

Comment:
Do these time zones have summer time? If so, we need to account for them.

Reply:
@eregon No, they don't have Daylight Saving Time. Re-checked.